common/src/main/scala/net/psforever/packet/game/SquadWaypointRequest.scala

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2019 PSForever
+package net.psforever.packet.game
+
+import net.psforever.packet.{GamePacketOpcode, Marshallable, PacketHelpers, PlanetSideGamePacket}
+import net.psforever.types.Vector3
+import scodec.{Attempt, Codec, Err}
+import scodec.codecs._
+import shapeless.{::, HNil}
+
+/**
+  * Actions that can be requested of the specific waypoint.
+  */
+object WaypointEventAction extends Enumeration {
+  type Type = Value
+
+  val
+  Add,
+  Unknown1,
+  Remove,
+  Unknown3 //unconfirmed
+  = Value
+
+  implicit val codec : Codec[WaypointEventAction.Value] = PacketHelpers.createEnumerationCodec(enum = this, uint2)
+}
+
+/**
+  * na
+  * @param zone_number the zone
+  * @param pos the continental map coordinate location of the waypoint;
+  *            the z-coordinate is almost always 0.0
+  */
+final case class WaypointInfo(zone_number : Int,
```

Comment:
Dont we have an enum for Zones?

Reply:
I have an ancient "put-on-hold" ticket related to this that is me griping that we have continents that are referenced by string name (internal name) and referenced by their ordinal number (sometimes an `Int` and sometimes `PlanetSideZoneID`) and referenced by their map name and the only thing that connects them is the heavy `Zone` object,  I have at least two functions covering for conversions between these references at a higher level.  It'd be easier to just put together an enum-ish container.